homeassistant/components/myq/cover.py

```diff
@@ -131,22 +123,23 @@ async def async_open_cover(self, **kwargs):
         try:
             wait_task = await self._device.open(wait_for_state=False)
         except MyQError as err:
-            _LOGGER.error(
-                "Opening of cover %s failed with error: %s", self._device.name, str(err)
-            )
-            return
+            # Write current state which would be closed to HASS
+            self.async_write_ha_state()
```

Comment:
```suggestion
```

If the action failed, the state should not have changed since the state hasn't been written yet.